src/state/ChatSlice.ts

```diff
@@ -20,46 +24,51 @@ export const useChatState = create<ChatSlice>((set, get) => ({
     errorMessage: "",
     chats: [],
     currentChatId: "",
-    updateCurrentChat: (id: string) => { set({ currentChatId: id }) },
+    currentChat: undefined,
+    setCurrentChatId: (id: string) => { set({ currentChatId: id, currentChat: undefined }) },
     fetchChats: async () => {
         set({ loading: true });
         try {
-            const res = await client.get("/Chat");
-            if (res.status >= 400) {
-                set({ errorMessage: res.statusText, loading: false })
-                return;
-            }
-            set({ success: true, chats: res.data, loading: false });
+            const res = await HttpRequest({ uri: "/Chat", method: RESTMethod.Get }) as AxiosResponse<unknown, unknown>;
```

Comment:
Add return type inside HttpRequest and don't use AxiosResponse<unknown, unknown> in each situations